### PR TITLE
TRITON-1880 backport imgapi channel support to node-sdc-clients v8.x

### PR DESCRIPTION
**Pull request by timfoster**
This was originally PR joyent/node-sdc-clients#18 as created by our gerrit->github automation, but that incorrectly made a PR targeting the 'master' branch rather than the new '8.x' branch. Please see that PR for the comments so far.

### CODE REVIEW

**Comment by trentm** [comment]
>  but that incorrectly made a PR targeting the 'master' branch rather than the new '8.x' branch.

Oops, sorry about that. :)   I think yours might have been the only CR ever to target non-master.

File: CHANGES.md [hunk 1]
```diff
@@ -5,11 +5,21 @@
...
 <!--
-    Copyright (c) 2014, Joyent, Inc.
+    Copyright 2019 Joyent, Inc.
 -->
 
 # sdc-clients Changelog
 
+## sdc-clients 8.2.0
+
+- Backport of channel support (TRITON-886) from v12.0.0
```
Comment (trentm) [inline CHANGES.md]: I think it could helpful to include the full changelog message from master. Perhaps:

```
- TRITON-886 Add support for `channel` option to IMGAPI client methods.
  Typically the "channel" is set as an argument to the IMGAPI client
  constructor. However, it can be useful to use an existing IMGAPI client
  to make a call for a separate channel.  (Backported from v12.0.0.)
```

File: lib/imgapi.js [hunk 2]
```diff
@@ -278,8 +278,8 @@ function IMGAPI(options) {
     } else {
         this.url = options.url;
     }
-    // _basePath: the URL subpath *without* a trailing '/'
-    this._basePath = parsed.pathname;
+    // the URL subpath *without* a trailing '/' or query params
+    this._basePath = parsed.path.split('?')[0];
```
Comment (trentm) [inline lib/imgapi.js]: Why this change?  My understanding is that `parsed.pathname` *is* a field without the query params:

```
> parsed = url.parse('https://images.joyent.com/foo/?channel=asdf')
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'images.joyent.com',
  port: null,
  hostname: 'images.joyent.com',
  hash: null,
  search: '?channel=asdf',
  query: 'channel=asdf',
  pathname: '/foo/',
  path: '/foo/?channel=asdf',
  href: 'https://images.joyent.com/foo/?channel=asdf' }
```

The current master lib/imgapi.js doesn't have this change.

File: lib/imgapi.js [hunk 3]
```diff
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
```
Comment (trentm) [inline lib/imgapi.js]: There was this hunk from commit 3d668a961ea2fff59afc41d78a555e02cf0e0c24

```
--- a/lib/imgapi.js
+++ b/lib/imgapi.js
@@ -9,7 +9,7 @@
  */

 /*
- * Client library for the SDC Image API (IMGAPI).
+ * Client library for the TritonDC Image API (IMGAPI).
  *
  * Usage without auth (e.g. when talking to in-SDC IMGAPI on admin network):
  *
```

Obviously minor, but any reason to not include it?

Comment (timfoster) [reply lib/imgapi.js:11]: Nope, no reason - we'll take that in.

File: CHANGES.md (hunk 1 above)
Comment (timfoster) [reply CHANGES.md:15]: Much better, thanks.

File: lib/imgapi.js (hunk 2 above)
Comment (timfoster) [reply lib/imgapi.js]: Yep, I went back and tested imgadm on SmartOS without this, and can't see why I had it originally. Reverted it now.
